src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -295,33 +296,21 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
     }
 
     /**
-     * Remove the outputStream which is receiving server output data
+     * Remove the outputStream which is receiving server output data.
      *
-     * @return true if the outputStream has been removed
+     * @return the outputStream has been removed if it is present
      */
-    public boolean removeOutputStream(OutputStream outputStream) {
+    public Optional<OutputStream> removeOutPutStream(OutputStream outputStream) {
         checkNotNull(outputStream, "outputStream parameter is NULL!");
         return stream.remove(outputStream);
     }
 
-    /**
-     * Remove the outputStream which is receiving server output data, and close
-     *
-     * @param outputStream
-     * @return true if the outputStream has been removed
-     */
-    public boolean removeOutputStreamAndClose(OutputStream outputStream) throws IOException {
-        boolean removed = removeOutputStream(outputStream);
-        outputStream.close();
-        return removed;
-    }
-
     /**
      * Close all existing server output streams.
      *
      * @throws IOException
      */
-    public void closeOutputStreams() throws IOException {
+    public void closeOutPutStreams() throws IOException {
```

Comment:
This method is also not needed now as we could always do something  similar to `removeOutPutStream(x).map(OutputStream::close)`